Quaver.Shared/Screens/Select/SelectScreen.cs

```diff
@@ -542,5 +573,184 @@ public void SelectRandomMap()
+                return;
+
+            switch (type)
+            {
+                case SelectContainerStatus.Mapsets:
+                    if (selectedMapset == null)
+                        return;
+
+                    filePath = Path.Combine(ConfigManager.SongDirectory.Value, selectedMapset.Directory);
+                    break;
+
+                case SelectContainerStatus.Difficulty:
+                    var selectedDifficulty = AvailableMapsets[view.MapsetScrollContainer.SelectedMapsetIndex].Maps[view.DifficultyScrollContainer.SelectedMapIndex];
+
+                    if (selectedDifficulty == null)
+                        return;
+
+                    filePath = selectedMapset.Maps.Count > 1 ? Path.Combine(Path.Combine(ConfigManager.SongDirectory.Value, selectedMapset.Directory), selectedDifficulty.Path) : Path.Combine(ConfigManager.SongDirectory.Value, selectedMapset.Directory);
+                    break;
+
+                default:
+                    throw new ArgumentOutOfRangeException();
+            }
+
+            var confirmDelete = new ConfirmCancelDialog($"Are you sure you want to delete this {( type == SelectContainerStatus.Mapsets ? "mapset" : type == SelectContainerStatus.Difficulty ? "difficulty" : null )} ? ", (sender, confirm) =>
+            {
+                MapDeletingInProgress = true;
+
+                try
+                {
+                    // Dispose of the background for the currently selected map.
+                    BackgroundHelper.Background?.Dispose();
+
+                    // Dispose of the currently playing track.
+                    AudioEngine.Track?.Dispose();
+
+                    // Run deletion & Reload in the background.
+                    ThreadScheduler.Run(() => DeleteAndReloadMaps(filePath));
+
+                    // Finally show confirmation notification
+                    NotificationManager.Show(NotificationLevel.Success, $"Successfully deleted {selectedMapset.Title} from Quaver!");
+                }
+                catch (ArgumentOutOfRangeException)
+                {
+                    // Handle error
+                    NotificationManager.Show(NotificationLevel.Error, "An error has occured. Please check runtime.log!");
+                }
+            });
+
+            // Finally show the confirmation dialog that orchestrates the deleting process.
+            DialogManager.Show(confirmDelete);
+        }
+
+        /// <summary>
+        ///     Used to delete a mapset or difficulty, then reload the available maps.
+        /// </summary>
+        /// <param name="path"></param>
+        private void DeleteAndReloadMaps(string path)
+        {
+            var view = View as SelectScreenView;
+            var type = view.ActiveContainer;
+
+            if (view.MapsetScrollContainer.SelectedMapsetIndex < 0)
+                return;
+
+            // Externally loaded map check
+            if (MapManager.Selected.Value.Game != MapGame.Quaver)
```

Comment:
Should be in the first function, no need to wait all the way until here to check this.

Reply:
Pretty sure @Swan said he wanted this here for reusability, not completely sure though.